src/game.cpp

```diff
@@ -11485,8 +11485,19 @@ bool game::walk_move( const tripoint &dest_loc )
     }
     u.set_underwater(false);
 
-    if( !shifting_furniture && !prompt_dangerous_tile( dest_loc ) ) {
-        return true;
+    bool fire_removed = false;
+    const tripoint furn_pos = u.pos() + u.grab_point;
+    const tripoint furn_dest = dest_loc + u.grab_point;
+    const int fire_str = m.get_field_strength( furn_pos, fd_fire );
+    const int fire_age = m.get_field_age( furn_pos, fd_fire );
+    if( m.get_field( furn_pos, fd_fire ) != nullptr || m.get_field_strength( furn_dest, fd_fire ) > 1 ) {
+        m.remove_field( furn_pos, fd_fire );
+        m.remove_field( furn_dest, fd_fire );
+        fire_removed = true;
+        if( shifting_furniture ) {
+            m.set_field_strength( furn_dest, fd_fire, fire_str );
+            m.set_field_age( furn_dest, fd_fire, fire_age );
+        }
```

Comment:
You don't need two different cases for moving the fire, though. In all cases where you move a stove containing a fire, you want to delete it at the start of the move, then recreate it at the end.